Corgibytes.Freshli.Cli/Functionality/Git/CachedGitSourceRepository.cs

```diff
@@ -32,21 +30,20 @@ public CachedGitSource FindOneByHash(string hash)
         return entry;
     }
 
+    public void Save(CachedGitSource cachedGitSource)
+    {
+        using var db = new CacheContext(Configuration.CacheDir);
+        db.CachedGitSources.Add(cachedGitSource);
+        db.SaveChanges();
+    }
+
     public CachedGitSource CloneOrPull(string url, string? branch)
     {
         // Ensure the cache directory is ready for use.
         CacheManager.Prepare();
 
         // Generate a unique hash for the repository based on its URL and branch.
-        using var sha256 = SHA256.Create();
-        var hashBytes = sha256.ComputeHash(Encoding.UTF8.GetBytes(url + branch));
-        var stringBuilder = new StringBuilder();
-        foreach (var hashByte in hashBytes)
-        {
-            stringBuilder.Append(hashByte.ToString("x2"));
-        }
-
-        var hash = stringBuilder.ToString();
+        var hash = new CachedGitSourceId(url, branch).Id;
```

Comment:
I like having this logic live in a class. Nice refactoring!

Reply:
Thank you! :D